src/main/scala/org/constellation/domain/checkpointBlock/CheckpointBlockBlacklistedAddressChecker.scala

```diff
@@ -0,0 +1,24 @@
+package org.constellation.domain.checkpointBlock
+
+import cats.effect.{Concurrent, Sync}
+import cats.implicits._
+import org.constellation.domain.blacklist.BlacklistedAddresses
+import org.constellation.primitives.{CheckpointBlock, Transaction}
+
+object CheckpointBlockBlacklistedAddressChecker {
+
+  def check[F[_]: Concurrent](cb: CheckpointBlock)(
+    blacklistedAddresses: BlacklistedAddresses[F]
+  ): F[List[Transaction]] =
+    cb.transactions.toList
+      .pure[F]
+      .flatMap(_.traverse(isFromBlockedAddress(_)(blacklistedAddresses)))
+      .map(_.flatten)
+
+  private def isFromBlockedAddress[F[_]: Concurrent](tx: Transaction)(
+    blacklistedAddresses: BlacklistedAddresses[F]
+  ): F[Option[Transaction]] =
+    blacklistedAddresses
+      .contains(tx.src.address)
+      .ifM(tx.some.pure[F], Sync[F].pure(None))
```

Comment:
can you change it to one format?
either `tx.some.pure[F], none[Transaction].pure[F]` (preferred?)
or `Sync[F].pure(Some(tx), Sync[F].pure(None)`,
or maybe `Sync[F].pure(tx.some), Sync[F].pure(none[Transaction])`